patina_dxe_core/src/lib.rs

```diff
@@ -670,60 +670,64 @@ mod tests {
         assert!(!<TestPlatform as MemoryInfo>::prioritize_32_bit_memory());
     }
 
+    fn with_clean_protocol_db<F>(f: F) -> bool
+    where
+        F: Fn() + std::panic::RefUnwindSafe,
+    {
+        test_support::with_global_lock(|| {
+            // SAFETY: init_test_protocol_db modifies global state. It is being called within a
+            // lock to have exclusive mutable access to the protocol database.
+            unsafe {
+                test_support::init_test_protocol_db();
+            }
+
+            f()
+        })
+        .is_err_and(|err| {
```

Comment:
This also doesn’t seem like it will always be valid. Right now the few tests here all end with calling bds, but, they could error otherwise, say before calling bds, and this would be confusing and other tests may be written that don’t call bds at all. I would recommend leaving the error handling in each test and just let them error out.